server/gitrest/Dockerfile

```diff
@@ -24,7 +25,8 @@ RUN corepack enable
 # Need to set the --unsafe-perm flag since we are doing the install as root. Consider adding an 'app' accout so we
 # can do the install as node but then switch to 'app' to run. As app we won't be able to write to installed files
 # and be able to change them.
-RUN pnpm install --unsafe-perm
+# Needs to set --no-frozen-lockfile since npmrc requires it if there are resolution changes, and the package dependencies don't exactly match
+RUN pnpm install --unsafe-perm --no-frozen-lockfile
```

Comment:
This change is bad, and should not be needed. Once https://github.com/microsoft/FluidFramework/pull/23330 is unblocked, and merged, this hack should be removed, then this PR can go forward